### PR TITLE
Improve code quality of CommandHistory

### PR DESCRIPTION
**Pull request by sheenkerr**
Deleted unused function in command history & added assertion and logging.

Also added comment in pinned person List Panel about the clicking outside not working (after a person has been selected)

### CODE REVIEW

**Review by ITLimJiaWei — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 1]
```diff
@@ -429,7 +429,7 @@ Furthermore, certain edits can cause the AddressBook to behave in unexpected way
 1. **When using multiple screens**, if you move the application to a secondary screen, and later switch to using only the primary screen, the GUI will open off-screen. The remedy is to delete the `preferences.json` file created by the application before running the application again.
 2. **If you minimize the Help Window** and then run the `help` command (or use the `Help` menu, or the keyboard shortcut `F1`) again, the original Help Window will remain minimized, and no new Help Window will appear. The remedy is to manually restore the minimized Help Window.
 3. **When you have extremely long tags**, the UI will not be able to display your tags properly. Text may not wrap, it may cause some parts to have scroll bars. Please avoid using excessive or really long tag names.
-
+4. Once a person is selected in either the pinned window or the contact list on the left, clicking on the empty area would not cause the currently selected person to change.
```
Comment (joeltio) [inline docs/UserGuide.md:432]: I think this is better suited to be in the DeveloperGuide as a known issue

**Review by joeltio — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (KennethTeo2002) [reply docs/UserGuide.md:432]: Would be better to rephrase it as "Once a person is selected, you are unable to unselect them from the ```ContactDetails``` panel"

**Review by joeltio — APPROVED** [review]
LGTM

**Review by KennethTeo2002 — APPROVED** [review]
LGTM

**Pull request merged**